src/main/java/org/opensearch/knn/plugin/script/KNNWhitelistExtension.java

```diff
@@ -10,17 +10,27 @@
 import org.opensearch.painless.spi.WhitelistLoader;
 import org.opensearch.script.ScoreScript;
 import org.opensearch.script.ScriptContext;
+import org.opensearch.script.ScriptedMetricAggContexts;
 
-import java.util.Collections;
 import java.util.List;
 import java.util.Map;
 
+import static java.util.Collections.singletonList;
+import static java.util.Map.entry;
+
 public class KNNWhitelistExtension implements PainlessExtension {
 
     private static final Whitelist WHITELIST = WhitelistLoader.loadFromResourceFiles(KNNWhitelistExtension.class, "knn_whitelist.txt");
 
     @Override
     public Map<ScriptContext<?>, List<Whitelist>> getContextWhitelists() {
-        return Collections.singletonMap(ScoreScript.CONTEXT, Collections.singletonList(WHITELIST));
+        List<Whitelist> whitelist = singletonList(WHITELIST);
+        return Map.ofEntries(
```

Comment:
Did you consider usage of ImmutableMap, it gives a bit more compact syntax, e.g.:
`ImmutableMap.of(ScoreScript.CONTEXT, whitelist)`

Reply:
Good point. I updated it now.